src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -154,12 +154,12 @@ internal static MarshalType GetMarshalTypeFromType(Type? type)
 
             switch (typeCode)
             {
-                case TypeCode.Byte:
                 case TypeCode.SByte:
                 case TypeCode.Int16:
-                case TypeCode.UInt16:
                 case TypeCode.Int32:
                     return MarshalType.INT;
+                case TypeCode.Byte:
+                case TypeCode.UInt16:
```

Comment:
we should make sure that this won't cause us to read past the end of a boxed value in the heap

Reply:
That means even more marshalers to implement, right ?